### DISCUSSION TITLE
[WASM + iOS] How to suppress 'bubbles'?

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#10931] [Q&A] [answered]
There are 'bubbles' of different sizes when using UNO WASM on a touch device (e.g. see [screenshots here](https://github.com/unoplatform/uno/issues/10902)). I suppose the purpose would be indicating where touch events occured...

Is there a way to suppress those 'bubbles'?

**Comment by jeromelaban** [comment, marked as answer]
Those are material press indicators, which are part of the templates provided by Uno.Material. You can override the templates to remove the "ripples". @kazo0 may have some additional details to give.

**Reply by DierkDroth** [reply]
Excellent! Thanks @jeromelaban. I'll give it a try...

**Reply by kazo0** [reply]
You can probably redefine the implicit style for the Ripple control that is being used in the Material styles, this could prevent you having to redefine the templates for each control. The default style should be [here](https://github.com/unoplatform/Uno.Themes/blob/master/src/library/Uno.Material/Styles/Controls/v2/Ripple.xaml), you can see an example of the Ripple control being used in places like the template of the [Material styled buttons](https://github.com/unoplatform/Uno.Themes/blob/1c2a0a0a2c53caf5d7b51d0f3cb32fba83e31017/src/library/Uno.Material/Styles/Controls/v2/Button.xaml#L152).

I would assume you could probably just create a new default implicit Style targeting `um:Ripple` that basically does nothing except provide a `Grid` or some other type of `Panel` as its template.

For reference, the Ripple control code is [here](https://github.com/unoplatform/Uno.Themes/blob/master/src/library/Uno.Material/Controls/Ripple.cs)

**Reply by DierkDroth** [reply]
Thanks @kazo0. I deleted the whole `VisualStateManager` in the Ripple template section which did the trick for me.
